coffee/shepherd.coffee

```diff
@@ -117,6 +117,9 @@ class Step extends Evented
     @tether = new Tether extend(tetherOpts, @options.tetherOptions)
 
   show: =>
+      @options.beforeShowPromise?()?.then(=> @_show()) ? @_show()
```

Comment:
```coffeescript
show: =>
  @options.beforeShowPromise?()?.then(@_show) ? @_show()

_show: =>
  ...
```

```coffeescript
tour = new Shepherd.Tour
tour.addStep 'start',
  title: 'Foo'
  text: 'bar'
  beforeShowPromise: ->
    p = Q.defer()
    setTimeout ->
      console.log('foo')
      p.resolve('foo')
    , 2000
    return p.promise

tour.start()
```

After starting the tour, it waits 2 seconds, logs 'foo', and then shows the tour. Commenting out the `beforeShowPromise` bit, it starts immediately. I happen to be using `Q` promises here, but it would work with anything that is thennable.